kube/Cargo.toml

```
@@ -16,57 +16,56 @@ categories = ["web-programming::http-client"]
 ws = ["tokio-tungstenite", "rand", "kube-core/ws"]
 oauth = ["tame-oauth"]
 gzip = ["async-compression"]
-admission = ["json-patch", "kube-core/admission"]
+admission = ["kube-core/admission"]
 
 [package.metadata.docs.rs]
-features = ["derive", "ws", "oauth", "jsonpatch"]
+features = ["derive", "client", "ws", "oauth", "jsonpatch", "admission"]
 # Define the configuration attribute `docsrs`. Used to enable `doc_cfg` feature.
 rustdoc-args = ["--cfg", "docsrs"]
 
 [dependencies]
-base64 = "0.13.0"
-chrono = "0.4.19"
-dirs = { package = "dirs-next", version = "2.0.0" }
+base64 = { version = "0.13.0", optional = true }
+chrono = { version = "0.4.19", optional = true }
+dirs = { package = "dirs-next", optional = true, version = "2.0.0" }
 serde = { version = "1.0.118", features = ["derive"] }
 serde_json = "1.0.61"
-serde_yaml = "0.8.17"
+serde_yaml = { version = "0.8.17", optional = true }
 http = "0.2.2"
 url = "2.2.0"
-log = "0.4.11"
-either = "1.6.1"
+either = { version = "1.6.1", optional = true }
 thiserror = "1.0.23"
-futures = "0.3.8"
-pem = "0.8.2"
+futures = { version = "0.3.8", optional = true }
+pem = { version = "0.8.2", optional = true }
 openssl = { version = "0.10.32", optional = true }
 tokio-native-tls = { version = "0.3.0", optional = true }
 tokio-rustls = { version = "0.22.0", features = ["dangerous_configuration"], optional = true }
-bytes = "1.0.0"
-tokio = { version = "1.0.1", features = ["time", "signal", "sync"] }
+bytes = { version = "1.0.0", optional = true }
+tokio = { version = "1.0.1", features = ["time", "signal", "sync"], optional = true }
 static_assertions = "1.1.0"
 kube-derive = { path = "../kube-derive", version = "^0.54.0", optional = true }
 kube-core = { path = "../kube-core", version = "^0.54.0"}
-jsonpath_lib = "0.2.6"
-tokio-util = { version = "0.6.0", features = ["io", "codec"] }
-json-patch = { version = "0.2.6", optional = true }
-hyper = { version = "0.14.2", features = ["client", "http1", "stream", "tcp"] }
+jsonpath_lib = { version = "0.2.6", optional = true }
+tokio-util = { version = "0.6.0", optional = true, features = ["io", "codec"] }
+hyper = { version = "0.14.2", optional = true, features = ["client", "http1", "stream", "tcp"] }
 hyper-tls = { version = "0.5.0", optional = true }
 hyper-rustls = { version = "0.22.1", optional = true }
 tokio-tungstenite = { version = "0.14.0", optional = true }
-tower = { version = "0.4.6", features = ["buffer", "util"] }
+tower = { version = "0.4.6", optional = true, features = ["buffer", "util"] }
 async-compression = { version = "0.3.7", features = ["gzip", "tokio"], optional = true }
-hyper-timeout = "0.4.1"
+hyper-timeout = {version = "0.4.1", optional = true }
 tame-oauth = { version = "0.4.7", features = ["gcp"], optional = true }
-pin-project = "1.0.4"
+pin-project = { version = "1.0.4", optional = true }
 rand = { version = "0.8.3", optional = true }
-tracing = "0.1.25"
+tracing = { version = "0.1.25", optional = true }
```

```suggestion
tracing = { version = "0.1.25", features = ["log"], optional = true }
```
I thought this was enabled by default, but it's not.
 
> `log`: causes trace instrumentation points to emit `log` records as well as trace events, if a default `tracing` subscriber has not been set. This is intended for use in libraries whose users may be using either `tracing` or `log`.
> https://docs.rs/tracing/0.1.26/tracing/#crate-feature-flags